src/java.base/share/classes/jdk/internal/lang/stable/TrustedFieldType.java

```diff
@@ -0,0 +1,18 @@
+package jdk.internal.lang.stable;
+
+import jdk.internal.lang.StableArray;
+import jdk.internal.lang.StableArray2D;
+import jdk.internal.lang.StableArray3D;
+import jdk.internal.lang.StableValue;
+
+/**
+ * This sealed marker interface signals fields that are _declared_ as a class that
+ * implements this interface is "trusted" and therefore somewhat protected from being
+ * modified via {@linkplain sun.misc.Unsafe} and {@linkplain java.lang.reflect.Field#setAccessible(boolean)}
+ * operations.
+ */
+public sealed interface TrustedFieldType
```

Comment:
Maybe export this interface to `jdk.unsupported`?

Reply:
> Maybe export this interface to `jdk.unsupported`?

I don't we should do that. In general, we need jdk.unsupported to go away in the long term. Also integrity of the platform depends on java.base being very stingy and not exporting internal packages to other modules.

Reply:
Given that `TrustedFieldType` is more generic than stable values, it could be moved to `jdk.internal.misc` or `jdk.internal.reflect`, then `jdk.unsupported` could use it without exporting new packages to `jdk.unsupported`.

Reply:
At some point in the future, 'jdk.unsupported' will be removed

Reply:
Maybe there is a better home for this?

Reply:
I don't think we should publish this API; this will soon be phased out by strict final fields (written only before super constructor calls) introduced by Valhalla, as strict final fields are never modifiable and can be safely trusted.

Reply:
Let's keep it internal.